pylint/checkers/base.py

```diff
@@ -1123,30 +1167,31 @@ def visit_with(self, node):
                         self.add_message('confusing-with-statement', node=node)
 
 
-_NAME_TYPES = {
-    'module': (MOD_NAME_RGX, 'module'),
-    'const': (CONST_NAME_RGX, 'constant'),
-    'class': (CLASS_NAME_RGX, 'class'),
-    'function': (DEFAULT_NAME_RGX, 'function'),
-    'method': (DEFAULT_NAME_RGX, 'method'),
-    'attr': (DEFAULT_NAME_RGX, 'attribute'),
-    'argument': (DEFAULT_NAME_RGX, 'argument'),
-    'variable': (DEFAULT_NAME_RGX, 'variable'),
-    'class_attribute': (CLASS_ATTRIBUTE_RGX, 'class attribute'),
-    'inlinevar': (COMP_VAR_RGX, 'inline iteration'),
+HUMAN_READABLE_TYPES = {
+    'module': 'module',
+    'const': 'constant',
+    'class': 'class',
+    'function': 'function',
+    'method':  'method',
+    'attr': 'attribute',
+    'argument': 'argument',
+    'variable': 'variable',
+    'class_attribute': 'class attribute',
+    'inlinevar': 'inline iteration'
 }
 
 def _create_naming_options():
     name_options = []
-    for name_type, (rgx, human_readable_name) in six.iteritems(_NAME_TYPES):
+    for name_type in DEFAULT_NAMING_STYLE.known_node_types():
+        human_readable_name = HUMAN_READABLE_TYPES[name_type]
         name_type = name_type.replace('_', '-')
         name_options.append((
             '%s-rgx' % (name_type,),
-            {'default': rgx, 'type': 'regexp', 'metavar': '<regexp>',
```

Comment:
Because these defaults would override every regular expressions taken from preset. Without removing these defaults, naming-style has no effect at all (condition in line 1285 is always True) . 

Now, when `--generate-rcfile` is used, naming-style returns user_friendly name, and all of custom regular expressions are included as placeholders. Seems fair enough for for both normal users and power users.